src/inmanta/util.py

```diff
@@ -445,7 +449,7 @@ def api_boundary_json_encoder(o: object) -> Union[ReturnTypes, "JSONSerializable
     """
     if isinstance(o, datetime.datetime):
         # Accross API boundaries, all naive datetime instances are assumed UTC. Returns ISO timestamp implicitly in UTC.
-        return datetime_utc_isoformat(o, naive_utc=True)
+        return datetime_iso_format(o, naive_utc=True)
```

Comment:
The current implementation still always uses naive times for the json encoder. This should be configurable as well.